back-end/src/main/java/index/Indexer.java

```diff
@@ -27,25 +28,38 @@
 
     // abstract methods
     public abstract void createMV(Canvas c, int layerId) throws Exception;
-    public abstract ArrayList<ArrayList<String>> getDataFromRegion(Canvas c, int layerId, String regionWKT, String predicate) throws Exception;
+    public ArrayList<ArrayList<String>> getDataFromRegion(Canvas c, int layerId, String regionWKT, String predicate) throws Exception {
```

Comment:
I don't think you can get rid of `abstract` here... it would essentially make Java ignore all old `getDataFromRegion` implementations of all child classes, and use this one in `Indexer.java`. I'm not very sure, but I think this will make other indexers break. 

Since what you proposed is the right interface (having both `oldbox` and `newbox` as arguments), I would just go with it, change everything and leave this function abstract (I can do it later).

Reply:
I was wrong - you can still get polymorphism by child classes overriding non-abstract method, like what you are doing here. 

However, I still prefer having one abstract method, since that enforces any child classes to implement this method (and that's the right interface!). Will make the changes accordingly.